### PR TITLE
feat: Add intentional nil encoding strategy to URLEncodedFormEncoder

### PR DESCRIPTION
**Pull request by darkbrewx**
## Issue Link 🔗

#3870 

## Goals ⚽

Enable different nil handling behaviors for `encode()` vs `encodeIfPresent()` in `URLEncodedFormEncoder`. Adds `.intentionalOnly` strategy that matches JSONEncoder semantics while maintaining full backward compatibility.

## Implementation Details 🚧

**Core Changes:**
- Extended `NilEncoding` with dual strategy support via `init(encode:encodeIfPresent:)` 
- Added `encodeNilIfPresent()` method and updated `KeyedContainer._encodeIfPresent()` to use it
- New `.intentionalOnly` strategy: encodes nil as "null" for `encode()`, skips for `encodeIfPresent()`

**Usage:**
```swift
let encoder = URLEncodedFormEncoder(nilEncoding: .intentionalOnly)

struct UserData: Encodable {
    let name: String? = nil
    let age: Int? = nil
    
    func encode(to encoder: any Encoder) throws {
        var container = encoder.container(keyedBy: CodingKeys.self)
        try container.encode(name, forKey: .name)        // → "name=null"  
        try container.encodeIfPresent(age, forKey: .age) // → skipped
    }
    
    enum CodingKeys: String, CodingKey { case name, age }
}
// Result: "name=null"
```

**Backward compatibility:** Existing `init(encoding:)` API unchanged.

## Testing Details 🔍

**New Tests:** 10 test cases in `IntentionalNilEncodingTests`
- ✅ `.intentionalOnly` strategy behavior
- ✅ Custom dual strategies  
- ✅ Array recursion and edge cases
- ✅ Backward compatibility

**Results:** All tests pass, existing URLEncodedFormEncoder tests unaffected.

### CODE REVIEW

File: Tests/IntentionalNilEncodingTests.swift [hunk 1]
```diff
@@ -0,0 +1,247 @@
+//
+//  IntentionalNilEncodingTests.swift
+//
+//  Copyright (c) 2014-2018 Alamofire Software Foundation (http://alamofire.org/)
```
Comment (jshier) [inline Tests/IntentionalNilEncodingTests.swift:4]: All new files should use the current year for copyright.

File: Tests/IntentionalNilEncodingTests.swift [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+//
+
+import Alamofire
+import Foundation
+import XCTest
+
+final class IntentionalNilEncodingTests: BaseTestCase {
+    // MARK: - Test Helper Structures
+
+    struct TestStruct: Encodable {
```
Comment (jshier) [inline Tests/IntentionalNilEncodingTests.swift:32]: Feel free to consolidate all of the test structs as `fileprivate`s at the bottom of this file.

File: Tests/IntentionalNilEncodingTests.swift (hunk 1 above)
Comment (jshier) [reply Tests/IntentionalNilEncodingTests.swift:4]: Really, though, these tests should be consolidated into the existing encoding tests.
